src/config.ml

```diff
@@ -135,6 +136,33 @@ module Conf_map = struct
   end
 
   include Gmap.Make(K)
+  let is_valid_server_config t =
+    let ensure_mem k err = if mem k t then Ok () else Error err in
+    let ensure_not k err = if not (mem k t) then Ok () else Error err in
+    let open Rresult in
+    R.reword_error (fun err -> `Msg ("not a valid server config: " ^  err))
+      ( ensure_mem Bind "does not have a bind" >>= fun()->
+        (match find Tls_mode t with
+         | None | Some `Client -> Error "is not a TLS server"
+         | Some `Server -> Ok ()) >>= fun () ->
+        let _todo = ensure_not in
+        begin match find Tls_cert t, find Tls_key t with
+          |  Some _, Some _ -> Ok ()
+          |  None, None ->
+            Error "does not have  TLS certificate and key"
+          | Some _, None -> 
+            Error "no tls certificate"
+          | None, Some _ ->
+            Error "no tls key"
+          (* ^-- TODO or has -pkcs12 *)
+        end >>= fun () ->
+        ensure_mem Cipher "server must specify 'cipher AES-256-CBC'"
+        >>= fun () ->
+        (if mem Cipher t && get Cipher t <> "AES-256-CBC" then
+           Error "currently only supported Cipher is 'AES-256-CBC'"
+         else Ok ()) 
+      )
+
```

Comment:
Wonder if it'd make sense to have a common function to check the common subset so we avoid duplication?